extide/gradle/arch.xml

```diff
@@ -268,6 +268,13 @@ publishPlugins&lt;/args&gt;
 &lt;/actions&gt;
              </pre>
          </api>
+         <api category="devel" group="java.io.File" name="nb-actions-config.xml" type="export">
+             <p>
+                 An extension of <a href="#nb-actions-config.xml">nb-actions.config</a> is a per-Configuration action mapping.
```

Comment:
I convinced Laszslo in the past to not expose `nbactions*.xml` files and rather encode the actions into `build.properties` file. The introduciton of `nb-actions-config.xml` goes against that decision. It is OK, if you and Laszlo are OK with that. I just have to mention it.

Reply:
This is a good topic for a discussion, I believe: at this point, the behaviour is +- consistent with Maven projects (which does not say anything about being valid). In a purist's view, the action mapping does not belong to the project shared metadata at all: 
- if an option (task, argument) is to be seen by the project itself, it should be put to the buildscript (pom) in a native way.
- if the project team agrees to share IDE-related metadata, it's best to have them file-separated, IMHO, from the project file(s) that are used by CI
- if the action mapping is not to be shared, it's best to put on `.gitignore` list; 

You're right that people frown upon `nbproject` and the like directories in VCS, but I see people routinely put their `.idea` directories to `.gitignore` lists. In a sense, its current impl that merges `gradle.properties` values that might affect CI with IDE-related setting may be also frowned upon. 

So I would stick to that pattern, and **document** to put 
```
**/nb-actions*.xml
**/nb-configuration*.xml
```
to `.gitignore` -- that would work for both Maven and Gradle.

@lkishalmi  -- what's your take on this ? I can +- easily change impl to persist all mappings to `gradle.properties` (using other prefixes); but that would eventually put there even customizations for private configurations (not sharable through `nb-configurations.xml`). I have no strong preference, as I can believe see benefits (or better consistency) of either solution, but slightly inclined to the separated files (although possibly in a subdir to limit the project pollution).

Re. this IDE-metadata as a general topic (should we create, should it be collected in a subdir, how to tackle private info - see [NETBEANS-5711](https://issues.apache.org/jira/browse/NETBEANS-5711): shouldn't it be debated on a mailing list instead ?

Reply:
For me it is kind of indifferent. I see value in both approaches. My day job is as DevOps engineer, I usually can make either way accepted in source repositories.
I'd open a discussion on the dev (maybe even the users) mailing list(s).